dmarc/src/main/java/org/apache/james/dmarc/DMARCVerifier.java

```diff
@@ -0,0 +1,137 @@
+/******************************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one                 *
+ * or more contributor license agreements.  See the NOTICE file               *
+ * distributed with this work for additional information                      *
+ * regarding copyright ownership.  The ASF licenses this file                 *
+ * to you under the Apache License, Version 2.0 (the                          *
+ * "License"); you may not use this file except in compliance                 *
+ * with the License.  You may obtain a copy of the License at                 *
+ *                                                                            *
+ *   http://www.apache.org/licenses/LICENSE-2.0                               *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing,                 *
+ * software distributed under the License is distributed on an                *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY                     *
+ * KIND, either express or implied.  See the License for the                  *
+ * specific language governing permissions and limitations                    *
+ * under the License.                                                         *
+ ******************************************************************************/
+package org.apache.james.dmarc;
+
+import org.apache.james.dmarc.exceptions.DmarcException;
+import org.apache.james.mime4j.dom.Message;
+
+public class DMARCVerifier {
+    public static final String FROM = "From";
+    private final String _dmarcResponse;
+    private final String _dmarcNonResponse;
+    private final PublicKeyRecordRetrieverDmarc _recordRetriever;
+
+    public DMARCVerifier(String dmarcResponse, String dmarcNonResponse, PublicKeyRecordRetrieverDmarc recordRetriever) {
+        _dmarcResponse = dmarcResponse;
+        _dmarcNonResponse = dmarcNonResponse;
+        _recordRetriever = recordRetriever;
+    }
+
+    public String runDmarcCheck(Message message, String spfHeaderText, String
```

Comment:
I would return a `DmarcValidationResult` POJO with a convenient `asString` method instead of just returning an unstructured String